### ISSUE TITLE
Kiribati (1.7.b) Edit vaccination event data

### ISSUE THREAD

**Issue by katherine-sussol** [openmsupply/mobile#4624]
## Is your feature request related to a problem? Please describe.

Update mSupply Mobile to allow the user to edit previously captured vaccination event data.
Previous versions of edited data will be kept in the back-end for auditing purposes.

Trello: https://trello.com/c/ZCGvqcx3/73-17b-edit-vaccination-event-data

## Describe the solution you'd like

- [ ] #4639 Need to create link between name_notes and json schema version used
- When name_note is edited, old one should be soft-deleted and retained in the backend instead of replacing 
- UI updates to allow editing
  - Step 1: #4691 
  - Step 2: #4714
  - Step 3: #4715

## Implementation

**A description of how the solution would best be implemented in the code base**

Some notes on the discussion of the approach to take here: https://docs.google.com/document/d/1ZO-41zCX0FJl41OiwRKdB7qGGUEkASecPzVCqVOzYLY/edit#

Summary: 

Issues:
- [x] #4635
- [x] #4638 
- [ ] #4636
- [x] #4637
- Delete transact.custom_data / stop using it
- When editing non-stock: shouldn’t need to cancel transaction??
- When editing stock: cancel transaction and create a new one (but make it seem like an edit to the user)
- User can’t edit stock if the transaction is not ‘local’ to their tablet - display message instead
- Note: If deleting the last vaccination event from a patient then probably need to make a new PCD without a transaction id

Current method of storing vaccination events: 

![2022-03-16_12-51-59](https://user-images.githubusercontent.com/19318937/158490747-68af4b8a-b97b-498c-8553-65927194093a.png)

Basically want to move to a new model where the transact.custom_data is stored in the Vax name_note. 

Some (possibly unhelpful) background info here: https://docs.google.com/presentation/d/1rS0hbZTa7iAnBIGWtForcLuBiInQiGEHM33HNHsS9kI/edit#slide=id.g1197003ea89_0_613

## Describe alternatives you've considered

**A clear and concise description of any alternative solutions or features you've considered.**

## Additional context

**Add any other context or screenshots about the feature request here.**

**Comment by adamdewey** [comment]
Discussed with @katherinesetiadarma , edit functionality looking really good. Some decisions to get it across the line:

- Re-arrange columns so 'Extra information' is 1st, then 'Additional Details' 2nd and 'Vaccination Event' 3rd
- Rename columns to 'Page 1', 'Page 2', 'Page 3'

**Comment by katherine-sussol** [comment]
3 Column layout referenced above is this monstrosity ('nice to have' but not doing just yet is to start the form out with everything disabled and make users explicitly click an 'Edit' button to enable the JSON schema form):
![image](https://user-images.githubusercontent.com/65875762/173260415-0f214257-3e22-4461-a9b3-245c0149eb15.png)

**Comment by adamdewey** [comment]
Nice Kat, just thinking - is it an easy fix change the colour on the toggle buttons, I still get confused about which one is selected, Yes or No! 

Maybe if both selected and not-selected were the same colour (grey or orange) it might be a bit clearer?

**Comment by katherine-sussol** [comment]
> Nice Kat, just thinking - is it an easy fix change the colour on the toggle buttons, I still get confused about which one is selected, Yes or No!
> 
> Maybe if both selected and not-selected were the same colour (grey or orange) it might be a bit clearer?

Oh, oops yes I think you mentioned this earlier and we didn't do anything about it (forgetful 🐱). Uhm. Something like this?
![image](https://user-images.githubusercontent.com/65875762/173267123-81028227-ab2b-4079-a5e0-e4622764b102.png)

**Comment by adamdewey** [comment]
Yes perfect! :)

**Comment by sworup** [comment]
@adamdewey @chastichno @nisha-dangol 
Edit is done and all of its issues has been merged. Except for #4639, but that is too big to be doable in first phase (in next few days). 

There are bugs and we should focus on the bugs, but this Epic can be closed as it main tasks has been merged.
